src/azure-cli/azure/cli/command_modules/synapse/operations/sqlpoolsecurityalertpolicy.py

```diff
@@ -0,0 +1,194 @@
+        # and then PUT it back. (We don't want the user to be force to specify this, because that
+        # would be very annoying when updating non-storage-related properties).
+        # This doesn't work if the user used generic update args, i.e. `--set state=Enabled`
+        # instead of `--state Enabled`, since the generic update args are applied after this custom
+        # function, but at least we tried.
+        if not storage_account:
+            storage_account = _get_storage_account_name(instance.storage_endpoint)
+            storage_resource_group = _find_storage_account_resource_group(cli_ctx, storage_account)
+
+        instance.storage_account_access_key = _get_storage_key(
+            cli_ctx,
+            storage_account,
+            storage_resource_group,
+            use_secondary_key)
+
+
+def _find_storage_account_resource_group(cli_ctx, name):
+    '''
+    Finds a storage account's resource group by querying ARM resource cache.
+
+    Why do we have to do this: so we know the resource group in order to later query the storage API
+    to determine the account's keys and endpoint. Why isn't this just a command line parameter:
+    because if it was a command line parameter then the customer would need to specify storage
+    resource group just to update some unrelated property, which is annoying and makes no sense to
+    the customer.
+    '''
+
+    storage_type = 'Microsoft.Storage/storageAccounts'
+    classic_storage_type = 'Microsoft.ClassicStorage/storageAccounts'
+
+    query = "name eq '{}' and (resourceType eq '{}' or resourceType eq '{}')".format(
+        name, storage_type, classic_storage_type)
+
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+    resources = list(client.resources.list(filter=query))
+
+    if not resources:
+        raise CLIError("No storage account with name '{}' was found.".format(name))
+
+    if len(resources) > 1:
+        raise CLIError("Multiple storage accounts with name '{}' were found.".format(name))
+
+    if resources[0].type == classic_storage_type:
+        raise CLIError("The storage account with name '{}' is a classic storage account which is"
+                       " not supported by this command. Use a non-classic storage account or"
+                       " specify storage endpoint and key instead.".format(name))
+
+    # Split the uri and return just the resource group
+    return resources[0].id.split('/')[4]
+
+
+def _get_storage_account_name(storage_endpoint):
+    '''
+    Determines storage account name from endpoint url string.
+    e.g. 'https://mystorage.blob.core.windows.net' -> 'mystorage'
+    '''
+    # url parse package has different names in Python 2 and 3. 'six' package works cross-version.
+    from six.moves.urllib.parse import urlparse  # pylint: disable=import-error
+
+    return urlparse(storage_endpoint).netloc.split('.')[0]
```

Comment:
Duplicate definition：
sqlpoolsecurityalertpolicy.py +140 
sqlpoolblobauditingpolicy.py +165

Reply:
fixed, removed the definition in this file and reuse the definition in `sqlpoolblobauditingpolicy.py` file

---
In reply to: [533160003](https://github.com/Azure/azure-cli/pull/16095#discussion_r533160003) [](ancestors = 533160003)